blocks/bricks/sequence_generators.py

```diff
@@ -299,12 +334,22 @@ def generate(self, outputs, **kwargs):
         states = dict_subset(kwargs, self._state_names)
         contexts = dict_subset(kwargs, self._context_names)
         glimpses = dict_subset(kwargs, self._glimpse_names)
+        lm_states = {}
+        if self.language_model:
+            lm_states = OrderedDict([(name[3:], state) for name, state in dict_subset(
+                kwargs, self._lm_state_names).items()])
+            lm_states = OrderedDict([('lm_' + name, state) for name, state
+                         in self.language_model.generate(
+                            outputs, as_dict=True, iterate=False,
+                            **lm_states).items()
+                         if name != 'costs'])
```

Comment:
No, it wouldn't because readout cannot accept it. But I still want to have outputs there because of FST transition, I return probabilities there as outputs. Maybe I should rethink it and create a special emitter for FSTs.